OneCommunityGlobal/HGNRest
# Shiwani add permission for edit project category or status

# Description
Make permission management fix for edit project category or status.

## Related PRS (if any):
To test this backend PR you need to checkout the [#1606](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1606) frontend PR.

## Main changes explained:
- Added condition for editProject permission to edit and update category and status for that project. The first condition for editProject checks if that permission is not present in frontPermissions and if it present then the if condition will not execute, but if the permission does not exist, it checks if the user has putProject permissions and if both of these permissions are not present then 403 error will be raised when trying to edit category and status.


## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. Follow steps in frontend PR

## Screenshots or videos of changes:

Before:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78906820/72ce48ed-e3c4-4366-a34d-64fc0d85059b



After:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78906820/65c5e668-b1d9-46e7-be2a-b5dcc1fa4ff9
